cms/djangoapps/contentstore/views/tests/test_item.py

```diff
@@ -1328,6 +1329,11 @@ def setUp(self):
         super(TestComponentTemplates, self).setUp()
         self.templates = get_component_templates(self.course)
 
+        # Initialize the deprecated modules settings with empty list
+        XBlockDeprecatedAdvancedComponentConfig.objects.create(
```

Comment:
`ConfigurationModel` objects are cached. It is possible that is causing whatever you are seeing. There is some logic in `ConfigurationModel.save()` which clears the cache. You can move that logic to a separate method `clear_cache()` and call it instead.